lib/posix/mqueue.c

```diff
@@ -51,7 +51,7 @@ mqd_t mq_open(const char *name, int oflags, ...)
 	va_list va;
 	mode_t mode;
 	mq_attr *attrs = NULL;
-	u32_t msg_size = 0, max_msgs = 0;
+	s32_t msg_size = 0, max_msgs = 0;
```

Comment:
why don't you use long here ? You are casting long to s32_t that still not right.